src/plugins/intel_gpu/src/plugin/transformations/swiglu_fusion.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "swiglu_fusion.hpp"
+
+#include "intel_gpu/op/swiglu.hpp"
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/swish.hpp"
+#include "openvino/op/variadic_split.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+SwiGLUFusion::SwiGLUFusion() {
+    using namespace ov::pass::pattern;
+
+    // Detect SwiGLU decomposition pattern
+    // SwiGLU(Xw, Xv, beta) = (Xw * (1.0 + exp(-beta * Xw))) * Xv
+    auto data = any_input();
+
+    // VariadicSplit(X, axis, split_lengths) = Xw, Xv
+    auto const_axis = wrap_type<ov::op::v0::Constant>();
+    auto const_split_lengths = wrap_type<ov::op::v0::Constant>();
+    auto variadic_split = wrap_type<ov::op::v1::VariadicSplit>({data, const_axis, const_split_lengths});
+
+    // Swish(Xw) = Xw * (1.0 + exp(-beta * Xw))
+    auto swish = wrap_type<ov::op::v4::Swish>({variadic_split});
+
+    // Mul(Xw, Xv) = Swish(Xw) * Xv
+    auto mul = wrap_type<ov::op::v1::Multiply>({swish, variadic_split});
+
+    ov::matcher_pass_callback callback = [=](ov::pass::pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        auto input = pattern_map.at(data);
+
+        auto vsplit = std::dynamic_pointer_cast<ov::op::v1::VariadicSplit>(pattern_map.at(variadic_split).get_node_shared_ptr());
```

Comment:
Please add transformation callback handling here

Reply:
Added transformation callback for matched root node